standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java

```diff
@@ -399,7 +449,8 @@ public List<CompactionInfo> findReadyToClean(long minOpenTxnWaterMark, long rete
          * By filtering on minOpenTxnWaterMark, we will only cleanup after every transaction is committed, that could see
          * the uncompacted deltas. This way the cleaner can clean up everything that was made obsolete by this compaction.
          */
-        String whereClause = " WHERE \"CQ_STATE\" = " + quoteChar(READY_FOR_CLEANING) + 
+        String whereClause = " WHERE \"CQ_STATE\" = " + quoteChar(READY_FOR_CLEANING) +
+          " AND \"CQ_TYPE\" != " + quoteChar(TxnStore.ABORT_CLEANUP_TYPE) +
```

Comment:
I think this should be in sync with COMPACTOR_CLEAN_ABORTS_USING_CLEANER value. if that is set to false this method should not filter out the aborts.

Reply:
I dont think this should be dependent on COMPACTOR_CLEAN_ABORTS_USING_CLEANER config because compaction should not pick the abort retry based records itself.